examples/aa-simple-dapp/src/components/Content.tsx

```diff
@@ -0,0 +1,326 @@
+"use client";
+import { AlchemyNFTAbi, alchNftContractAddress } from "@/config/alch-nft";
+import { AlchemyTokenAbi, alchTokenContractAddress } from "@/config/alch-token";
+import { useWalletContext } from "@/context/wallet";
+import { PluginType } from "@/hooks/usePluginManager";
+import { getPluginTypeFromAddress } from "@/shared/util";
+import { MSCA, SessionKeyPlugin, type Plugin } from "@alchemy/aa-accounts";
+import { UserOperationCallData } from "@alchemy/aa-core";
+import { useCallback, useState } from "react";
+import { Address, Hash, encodeFunctionData } from "viem";
+
+type MintStatus =
+  | "Mint"
+  | "Requesting"
+  | "Minting"
+  | "Received"
+  | "Error Minting";
+
+type PluginUoStatus = "Requesting" | "In Progress..." | "Done" | "Error";
+
+export default function Content() {
+  const {
+    isLoggedIn,
+    provider,
+    scaAddress,
+    installedPlugins,
+    availablePlugins,
+    pluginInstall,
+    pluginUninstall,
+    refetchInstalledPlugins,
+    tokenReceiverEnabled,
+    sessionKey,
+    sessionKeySigner,
+    sessionKeyEnabled,
+  } = useWalletContext();
+  const [mintTxHash, setMintTxHash] = useState<Hash>();
+  const [mintStatus, setMintStatus] = useState<MintStatus>("Mint");
+  const [mintType, setMintType] = useState<"erc20" | "erc721" | null>(null);
+
+  const [pluginTxHash, setPluginTxHash] = useState<Hash>();
+  const [pluginTxStatus, setPluginTxStatus] = useState<PluginUoStatus | null>(
+    null
+  );
+  const [pluginType, setPluginType] = useState<PluginType | null>(null);
+
+  const handleMint = useCallback(
+    async (nft: boolean = false) => {
+      if (!provider) {
+        throw new Error("Provider not initialized");
+      }
+
+      if (nft && !tokenReceiverEnabled) {
+        throw new Error("Account requires TokenReceiverPlugin installed first");
+      }
+
+      try {
+        setMintTxHash(undefined);
+        setMintStatus("Requesting");
+        setMintType(nft ? "erc721" : "erc20");
+        const userOpData: UserOperationCallData = {
+          target: nft ? alchNftContractAddress : alchTokenContractAddress,
+          data: encodeFunctionData({
+            abi: nft ? AlchemyNFTAbi : AlchemyTokenAbi,
+            functionName: nft ? "mintTo" : "mint",
+            args: [await provider.getAddress()],
+          }),
+          value: BigInt(0),
+        };
+
+        const executeWithSessionKey =
+          provider.isConnected<MSCA>() && sessionKeyEnabled && sessionKey;
+        console.log("executing with session key", executeWithSessionKey);
+        const uoHash = executeWithSessionKey
+          ? await provider
+              .extend(SessionKeyPlugin.providerMethods)
+              .executeWithSessionKey(
+                {
+                  args: [
+                    [{ ...userOpData, value: userOpData.value || BigInt(0) }],
+                    sessionKey,
+                  ],
+                },
+                {
+                  signer: sessionKeySigner,
```

Comment:
yea I'm still not seeing the use case for this... there's a method in the `WalletContext` (i think?) called `connectProviderToAccount` which takes in a signer and an address. Why not use that method like so:

```ts
const { provider, connectProviderToAccount } = useWalletContext();
const { sessionKeySigner } = useSessionKey();

if (sessionKeySigner && provider.isConnected<MSCA>()) {
  connectProviderToAccount(sessionKeySigner, await provider.getAddress());
}
```

Now you don't need to override the signer ever

Reply:
you could probably have this `connectProviderToAccount` logic in the WalletContext itself (maybe in the `login` method or method that checks magic's existing state)

Reply:
I don't think this is a better approach.

1. Then the provider will only be connected to session key signer which is not an MSCA. then typescript for MSCA breaks because provider is not `Provider & { account?: MSCA }` (maybe it won't, haven't checked)
2. we might want to add a fallback to original account with signing in case session key signing somehow goes wrong. 

Also, I want to discuss about making `Plugin` as an object so that we can do `account.plugins.sessionkey.<some plugin method>` ultimately instead of using Static calls because currently devs have to manage individual plugin code all separately. Gonna be end up writing react hooks for every single plugin being used in their projects.

Reply:
3. we also need to support all other overrides still for session key enabled accounts
4. we still want the original account to be connected to the provider to not let users bypass gas policy if using paymaster. We can still do that but then devs have to set the accountAddress for the smart account being connected, and that is quite hidden for devs. I can see devs often just copying pasting the original connect code here for session keys and accidentally have the session key signer being connected to another account address. There is no technical good way to catch these mistakes.

Reply:
One way to solve 2 (which will solve 4 as well) is to define the concept of a `SessionKeySigner` which can handle the fallback logic. That way when you connect an account you can connect it with the SessionKeySigner off the bat.

By consolidating logic into one class for signing, this should keep it in line with the rest of the flows we have. I'm weary of how many overrides we have now and it's not the best pattern for everything.